### ISSUE TITLE
[DataGrid] Accessibility aria-colindex  should start at 1

### ISSUE THREAD

**Issue by marnibrewster** [mui-org/material-ui-x#1657]
Mui DataGrid is failing an axe accessibility audit for creating `aria-colindex` starting at 0 instead of 1. 

`GridCell` sets `aria-colindex` to the `colIndex` passed in as a prop. `GridColumnHeaderItem` sets `aria-colindex` to `colIndex + 1` which I _believe_ is the preferred solution. 

Here is the documentation to explain that the numbers must start at 1 and increment: https://www.digitala11y.com/aria-colindexproperties/

I ran an axe audit (using the Chrome extension) on the most basic DataGrid demo and it failed, so I am fairly certain this is not an error I introduced.

- [x ] The issue is present in the latest release.
- [x ] I have searched the [issues](https://github.com/mui-org/material-ui-x/issues) of this repository and believe that this is not a duplicate.

## Current Behavior 😯

`GridPanel` cells are given an `aria-colindex` starting at 0.

## Expected Behavior 🤔

`GridPanel` cells should be given an `aria-colindex` starting at 1. 

## Steps to Reproduce 🕹

Steps:

1. Open sandbox: https://codesandbox.io/s/kind-tu-h2g8g?file=/src/Demo.tsx
2. Inspect the first row, first cell
2. Look at the `aria-colindex` and see that it equals 0:
![Screen Shot 2021-05-12 at 4 19 16 PM](https://user-images.githubusercontent.com/13192976/118040503-993aa100-b33f-11eb-9958-0d8528062e20.png)

## Context 🔦
I'm aiming for not having any critical accessibility issues. This one seems like it might be relatively easy fix (updating the component and relevant tests). 

## Your Environment 🌎

<details>
  <summary>`npx @material-ui/envinfo`</summary>
  
```
   System:
    OS: macOS 10.15.7
  Binaries:
    Node: 14.15.4 - /usr/local/bin/node
    Yarn: Not Found
    npm: 6.14.10 - /usr/local/bin/npm
  Browsers:
    Chrome: 90.0.4430.212
    Edge: 90.0.818.56
    Firefox: Not Found
    Safari: 13.1.3
  npmPackages:
    @material-ui/core: ^4.9.9 => 4.11.0 
    @material-ui/data-grid: ^4.0.0-alpha.22 => 4.0.0-alpha.22 
    @material-ui/icons: ^4.9.1 => 4.9.1 
    @material-ui/pickers: ^3.2.10 => 3.2.10 
    @material-ui/styles:  4.10.0 
    @material-ui/system:  4.9.14 
    @material-ui/types:  5.1.0 
    @material-ui/utils:  4.10.2 
    @types/react:  16.9.51 
    react: ^16.8.6 => 16.13.1 
    react-dom: ^16.8.2 => 16.13.1 
    typescript: ^3.4.3 => 3.9.7 
```
</details>

## Order id 💳
null 😅

**Comment by dtassone** [comment]
```diff
-       aria-colindex={colIndex}
+      aria-colindex={colIndex +1}
```

**Comment by oliviertassinari** [comment]
1. Linking something on HEAD might work today but will fail tomorrow after a refactoring. Prefer using permalink:

https://github.com/mui-org/material-ui-x/blob/358a1c4ac22e01551ce8490345ac24f095ea95e8/packages/grid/_modules_/grid/components/cell/GridCell.tsx#L169



2. it's even easier with a git diff that developer can `git apply`:

```diff
diff --git a/packages/grid/_modules_/grid/components/cell/GridCell.tsx b/packages/grid/_modules_/grid/components/cell/GridCell.tsx
index b273da9a..2f76a759 100644
--- a/packages/grid/_modules_/grid/components/cell/GridCell.tsx
+++ b/packages/grid/_modules_/grid/components/cell/GridCell.tsx
@@ -166,7 +166,7 @@ export const GridCell = React.memo((props: GridCellProps) => {
       data-rowselected={isSelected}
       data-editable={isEditable}
       data-mode={cellMode}
-      aria-colindex={colIndex}
+      aria-colindex={colIndex + 1}
       style={style}
       tabIndex={tabIndex}
       {...eventsHandlers}
```

3. I wonder if we shouldn't fix the colIndex computation sooner in the pipeline, e.g. 

```diff
diff --git a/packages/grid/_modules_/grid/components/cell/GridRowCells.tsx b/packages/grid/_modules_/grid/components/cell/GridRowCells.tsx
index 4b1c1239..a56971db 100644
--- a/packages/grid/_modules_/grid/components/cell/GridRowCells.tsx
+++ b/packages/grid/_modules_/grid/components/cell/GridRowCells.tsx
@@ -60,7 +60,7 @@ export const GridRowCells = React.memo((props: RowCellsProps) => {
   const rowHeight = useGridSelector(apiRef, gridDensityRowHeightSelector);

   const cellsProps = columns.slice(firstColIdx, lastColIdx + 1).map((column, colIdx) => {
-    const colIndex = firstColIdx + colIdx;
+    const colIndex = firstColIdx + colIdx + 1;
     const isLastColumn = colIndex === columns.length - 1;
     const removeLastBorderRight = isLastColumn && hasScrollX && !hasScrollY;
     const showRightBorder = !isLastColumn
```

**Comment by dtassone** [comment]
3. no as the colIndex is ok and should be 0 based. It is just the `aria-colindex` that is 1 based

**Comment by oliviertassinari** [comment]
> colIndex is ok and should be 0 based

Regarding the why, I guess it's as a convention to make it simpler to reason about the different indexes.

**Comment by dtassone** [comment]
> > colIndex is ok and should be 0 based
> 
> Regarding the why, I guess 0-based is a convention to make it simpler to reason about the different indexes.

The index is 0 based because it is the index position in the array of columns. As always, arrays starts at 0.

Aria-colindex and aria-rowindex are one-based as they're used by screen readers.

https://www.w3.org/TR/wai-aria-practices/examples/grid/dataGrids.html
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Gridcell_role
